src/docfx/restore/RestoreGit.cs

```diff
@@ -143,6 +143,9 @@ void AddWorkTrees()
                                         // it may be dirty caused by last failed restore action
                                         if (Directory.Exists(workTreePath))
                                             Directory.Delete(workTreePath, true);
+
+                                        // https://stackoverflow.com/questions/24265481/after-directory-delete-the-directory-exists-returning-true-sometimes
+                                        Debug.Assert(!Directory.Exists(workTreePath));
```

Comment:
This only fails on debug, shall we log an exception for the normal run as well?

Reply:
this is only to identify the root cause, once we know it's the root cause, we will something here.